### PR TITLE
dell-dock: add usb protocol for USB4 device

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@CragW I fixed up a few cosmetic things and dropped fu_dell_dock_usb4_write_u32to_buf() -- can you check it still works and if so please click the "Rebase and merge" button. Thanks!

File: plugins/dell-dock/fu-dell-dock-usb-usb4.c [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+	guint32 bytes_total = length;
+
+	if (nvm_addr % 4) {
+		g_prefix_error (error,
+			       "Invalid NVM write offset 0x%x, must be DW aligned: ",
+			        nvm_addr);
+		return FALSE;
+	}
+	if (length < 64 || length % 64) {
+		g_prefix_error (error,
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-usb-usb4.c:369]: should be `g_set_error`

File: plugins/dell-dock/fu-dell-dock-usb-usb4.c [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+fu_dell_dock_usb4_hub_nvm_write (FuDevice *device, const guint8 *buf,
+				 guint32 length, guint32 nvm_addr,
+				 GError **error)
+{
+	guint8 metadata[4];
+	guint32 bytes_done = 0;
+	guint32 bytes_total = length;
+
+	if (nvm_addr % 4) {
+		g_prefix_error (error,
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-usb-usb4.c:363]: should be `g_set_error`

**Review by superm1 — APPROVED** [review]
I just see two instances that should be fixed up for errors, and then LGTM.  Fix them and merge.

File: plugins/dell-dock/fu-dell-dock-usb-usb4.c (hunk 1 above)
Comment (hughsie) [reply plugins/dell-dock/fu-dell-dock-usb-usb4.c]: Fixed, thanks.

File: plugins/dell-dock/fu-dell-dock-usb-usb4.c (hunk 2 above)
Comment (hughsie) [reply plugins/dell-dock/fu-dell-dock-usb-usb4.c]: Fixed, thanks.

**Review by superm1 — APPROVED** [review]

**Comment by CragW** [comment]
> @CragW I fixed up a few cosmetic things and dropped fu_dell_dock_usb4_write_u32to_buf() -- can you check it still works and if so please click the "Rebase and merge" button. Thanks!

It works. :)  Thank you!

**Pull request merged**